tests/sealing/enc/enc.c

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/enclave.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+#include <openenclave/corelibc/stdlib.h>
+#include <string.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include "sealing_t.h"
+
+oe_result_t seal_data(
+    oe_seal_policy_t policy,
+    const char* plaintext,
+    const char* aad,
+    uint8_t* blob,
+    size_t blob_max_size,
+    size_t* blob_size)
+{
+    oe_result_t result = OE_OK;
+    uint8_t* key_info = NULL;
+    size_t ki_size = 0;
+    uint8_t* b = NULL;
+    size_t b_size;
+
+    OE_CHECK(oe_get_seal_key_info(policy, NULL, 0, 0, &key_info, &ki_size));
+
+    OE_CHECK(oe_seal(key_info, ki_size,
+                     (const uint8_t*)plaintext, strlen(plaintext),
+                     (const uint8_t*)aad, strlen(aad),
+                     &b, &b_size));
+
+    if (b_size > blob_max_size)
+        result = OE_BUFFER_TOO_SMALL;
+    else
+    {
+        OE_CHECK(oe_memcpy_s(blob, blob_max_size, b, b_size));
+        *blob_size = b_size;
+    }
+
+done:
+    if (key_info != NULL)
+        oe_free_seal_key(NULL, key_info);
```

Comment:
As noted earlier this line is confusing... why is it not simply oe_free(key_info) ?

Reply:
Isn't `key_info` supposed to be freed by `oe_free_seal_key()` or `oe_free_key()`? If not, why do both APIs have a `key_info` parameter?